varianteval/scripts/README

```diff
@@ -1 +1,62 @@
-Utility scripts
+## Utility scripts
+### annotation_script.py
+Script to add annotation to input callset with respect to specified comparison callsets. Annotations identify
+records in the input callset that overlap with records in the comparison callsets. An example call is of the form:
+```
+python annotation_script.py --output_dir {path} --source_vcf {path} --comparison_vcfs {path1 path2 ...} --comparison_labels {label1 label2 ...}
+```
+The full set of input flags and default values are given here:
+```
+  --output_dir OUTPUT_DIR
+                        output directory for annotated files
+  --source_vcf SOURCE_VCF
+                        input vcf that will have annotations add to (a copy
+                        of) it
+  --comparison_vcfs COMPARISON_VCFS [COMPARISON_VCFS ...]
+                        list of comparison/conflict databases
+  --comparison_labels COMPARISON_LABELS [COMPARISON_LABELS ...]
+                        list of annotation labels corresponding to the input
+                        comparison vcfs
+  --conflict_overlap_thresh CONFLICT_OVERLAP_THRESH
+                        Overlap threshold to be applied when looking for
+                        conflicting calls of different type
+  --overlap_bp_margin OVERLAP_BP_MARGIN
+                        breakpoint margin to be applied in determining overlap
+                        of two intervals
+  --allowed_overlap_type {agnostic,same,different}
+                        SV type constraints on what events are considered
+                        overlapping: if 'agnostic' then collisions will be
+                        allowed across type; if 'same' then collisions will
+                        require events of the same type; if 'different' then
+                        collisions will require events of different type
+```
+
+### upset_from_annotations.py
+Script to parse an annotated VCF (with annotations indicating rows that are found in comparison callsets) and generate
+binary overlap matrices, and upsetplots from those. Assumes vcfs that are given are generated from SURVIVOR (i.e.,
+include SUPP_VEC value in the info field). An example call is of the form:
+```
+python upset_from_annotations.py --annot_vcf {path to annotated vcf} --output_dir {path} --tools {toolname1 toolname2 ...} --fig {path} --plot_type {type}
```

Comment:
does "annotations" here refer to the survivor support vector only? if so, we should rename "annotation" terminology to "survivor"

Reply:
The survivor support vector is required to make the upsetplot at all, and the annotations are required to create the stacked bars -- I edited the readme to explain that